src/main/java/com/gmail/nossr50/util/Misc.java

```diff
@@ -128,6 +128,20 @@ public static void spawnItemsFromCollection(@Nullable Player player, @NotNull Lo
         }
     }
 
+    /**
+     * Drops the item from the item stack only if it is a sapling (or equivalent)
+     * Needed for TreeFeller
+     */
+    public static void spawnItemIfSapling(@NotNull Player player, @NotNull Location location, @NotNull Collection < ItemStack > drops, @NotNull ItemSpawnReason itemSpawnReason) {
```

Comment:
A few notes...
1. What does passing `player` in contribute to this method? (actually, what is it doing in `spawnItem`?)
2. I would rename `location` to `spawnLocation` for clarity
3. Remove the extra whitespace in `< ItemStack >`
4. This method could be made more generic, for example something like this
```conditionallySpawn(@NotNull Predicate<String> predicate, @NotNull Location spawnLocation, @NotNull Collection<ItemStack> items, @NotNull ItemSpawnReason itemSpawnReason)```

Reply:
1) it's in there because I need to pass it to spawnItem.   It appears to be in spawnItem because you use it in the McMMOItemSpawnEvent constructor on line 172

2) Renamed.  It was location to match your spawnItem method.

3) Done.

4). Will figure out how to do this and resubmit.  It seemed overly specific as written.  Ty.

Reply:
4) now done